### PR TITLE
Support exec plugin on Windows

### PR DESCRIPTION
**Pull request by gochist**
As discussed on #2924, we are now aware that the exec plugin is not available on Windows. This is because we can't check if the plugin is executable via file mode on Windows.

So, I changed the conditions for loading exec plugin so that the plugin can be used in Windows.

### CODE REVIEW

File: api/internal/plugins/execplugin/execplugin.go [hunk 1]
```diff
@@ -50,7 +51,9 @@ func (p *ExecPlugin) ErrIfNotExecutable() error {
 	if err != nil {
 		return err
 	}
-	if f.Mode()&0111 == 0000 {
+	// In Windows, it is not possible to determine whether a
+	// file is executable through file mode.
+	if f.Mode()&0111 == 0000 && runtime.GOOS != "windows" {
```
Comment (Shell32-Natsu) [inline api/internal/plugins/execplugin/execplugin.go:56]: Can you add a test for this behavior?

Comment (gochist) [reply api/internal/plugins/execplugin/execplugin.go:56]: @Shell32-Natsu Thank you for the review. I'm not good at Go development, but I tried to add a test. I tried to skip the test in Windows environment, what do you think?

File: api/internal/plugins/execplugin/execplugin_test.go [hunk 2]
```diff
@@ -89,3 +92,32 @@ metadata:
 		t.Fatalf("unexpected arg array: %#v", p.Args())
 	}
 }
+
+func TestExecPlugin_ErrIfNotExecutable(t *testing.T) {
```
Comment (Shell32-Natsu) [inline api/internal/plugins/execplugin/execplugin_test.go:96]: @monopole Could you take a look. I think this test should work but not sure about the practice that skip it on Windows.

File: api/internal/plugins/execplugin/execplugin.go (hunk 1 above)
Comment (monopole) [reply api/internal/plugins/execplugin/execplugin.go:56]: @gochist To line 55, can you add this:

// TODO: provide for setting the executable bit on Windows.
// The (fs *fileStat) Mode() (m FileMode) {} function in
// in https://golang.org/src/os/types_windows.go
// lacks the ability to set the executable bit.

Comment (monopole) [reply api/internal/plugins/execplugin/execplugin.go:56]: then i think we should pull it in, and unblock windows use.

File: api/internal/plugins/execplugin/execplugin_test.go (hunk 2 above)
Comment (monopole) [reply api/internal/plugins/execplugin/execplugin_test.go:96]: yes, this is ok in this case.

File: api/internal/plugins/execplugin/execplugin.go (hunk 1 above)
Comment (gochist) [reply api/internal/plugins/execplugin/execplugin.go:61]: @monopole Thank you. I've addressed your review. And I will open an issue in the golang repo within a few days.

Comment (gochist) [reply api/internal/plugins/execplugin/execplugin.go:61]: @monopole I've opened an issue, https://github.com/golang/go/issues/41809 in the golang repo.

**Review by monopole — APPROVED** [review]
/lgtm
/ok-to-test

**Pull request merged**